synapse/handlers/room.py

```diff
@@ -172,7 +172,7 @@ async def upgrade_room(
         user_id = requester.user.to_string()
 
         # Check if this room is already being upgraded by another person
-        for key in self._upgrade_response_cache.pending_result_cache:
+        for key in self._upgrade_response_cache.keys():
```

Comment:
this is the only reference to `ResponseCache.pending_result_cache` outside `ResponseCache` itself.